### PR TITLE
crdclient: support Apply operations

### PR DESCRIPTION
**Pull request by stevenctl**
RE: #44165. In general we probably want SSA instead of Patch.

### CODE REVIEW

File: pilot/pkg/config/kube/crdclient/types.gen.go [hunk 1]
```diff
@@ -724,6 +730,83 @@ func patch(c kube.Client, orig config.Config, origMeta metav1.ObjectMeta, mod co
...
+		}, metav1.ApplyOptions{FieldManager: fieldManager, Force: force})
+	case gvk.VirtualService:
+		return c.Istio().NetworkingV1alpha3().VirtualServices(cfg.Namespace).Apply(context.TODO(), &applyconfigistioioapinetworkingv1alpha3.VirtualServiceApplyConfiguration{
+			ObjectMetaApplyConfiguration: getApplyConfigMetadata(objMeta),
+			Spec:                         cfg.Spec.(*istioioapinetworkingv1alpha3.VirtualService),
+		}, metav1.ApplyOptions{FieldManager: fieldManager, Force: force})
+	case gvk.WasmPlugin:
+		return c.Istio().ExtensionsV1alpha1().WasmPlugins(cfg.Namespace).Apply(context.TODO(), &applyconfigistioioapiextensionsv1alpha1.WasmPluginApplyConfiguration{
+			ObjectMetaApplyConfiguration: getApplyConfigMetadata(objMeta),
+			Spec:                         cfg.Spec.(*istioioapiextensionsv1alpha1.WasmPlugin),
```
Comment (howardjohn) [inline pilot/pkg/config/kube/crdclient/types.gen.go:793]: Doesn't the spec need to be ApplyConfiguration as well?

**Comment by stevenctl** [comment]
https://github.com/kubernetes-sigs/controller-runtime/issues/1669

oh...

File: pilot/pkg/config/kube/crdclient/types.gen.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/config/kube/crdclient/types.gen.go:793]: I think that might mean our client-go is broken.

Or our structs are unique relative to k8s core types so we don't need it possibly.

The reason its needed, btw, is to distinguish unset vs default. So I don't think we could apply something like `action=ALLOW` (default enum) for authz policy

Comment (stevenctl) [reply pilot/pkg/config/kube/crdclient/types.gen.go:793]: https://github.com/kubernetes-sigs/controller-tools/pull/536

Looks like this won't be possible using the k8s client for a bit.

**Pull request closed without merging**